src/main/java/slimeknights/tconstruct/smeltery/tileentity/FaucetTileEntity.java

```diff
@@ -0,0 +1,231 @@
+    if (isPouring) {
+      stopPouring = true;
+      return;
+    }
+
+    direction = getWorld().getBlockState(pos).get(FACING);
+    doTransfer();
+    return;
+  }
+
+  public void handleRedstone(boolean hasSignal) {
+    if (hasSignal != lastRedstoneState) {
+      lastRedstoneState = hasSignal;
+      if (hasSignal) {
+        getWorld().getPendingBlockTicks().scheduleTick(pos, this.getBlockState().getBlock(), 2);
+      }
+    }
+  }
+
+  @Override
+  public void tick() {
+    if (getWorld().isRemote) {
+      return;
+    }
+
+    // nothing to do if not pouring
+    if (!isPouring) {
+      return;
+    }
+
+    if (drained != null) {
+      // done draining
+      if (drained.getAmount() <= 0) {
+        drained = null;
+        // pour me another, if we want to
+        if (!stopPouring) {
+          doTransfer();
+        }
+        else {
+          reset();
+        }
+      }
+      else {
+        // reduce amount (cooldown)
+        pour();
+      }
+    }
+  }
+
+  protected void doTransfer() {
+    // still got content left
+    IFluidHandler toDrain = getFluidHandler(pos.offset(direction), direction.getOpposite());
+    IFluidHandler toFill = getFluidHandler(pos.down(), Direction.UP);
+    if (toDrain != null && toFill != null) {
+      // can we drain?
+      FluidStack drained = toDrain.drain(144, IFluidHandler.FluidAction.SIMULATE);
+      if (!drained.isEmpty()) {
+        // can we fill
+        int filled = toFill.fill(drained, IFluidHandler.FluidAction.SIMULATE);
+        if (filled > 0) {
+          // drain the liquid and transfer it, buffer the amount for delay
+          this.drained = toDrain.drain(filled, IFluidHandler.FluidAction.EXECUTE);
+          this.isPouring = true;
+          pour();
+
+          // sync to clients
+          if (!getWorld().isRemote()) { // TODO: getWorld() instanceof WorldServer
+            // TODO: TinkerNetwork.sendToClients((WorldServer) getWorld(), pos, new FaucetActivationPacket(pos, drained));
+          }
+
+          return;
+        }
+      }
+    }
+    // draining unsuccessful
+    reset();
+  }
+
+  protected void pour() {
+    if (drained.isEmpty()) {
+      return;
+    }
+
+    IFluidHandler toFill = getFluidHandler(pos.down(), Direction.UP);
+    if (toFill != null) {
+      FluidStack fillStack = drained.copy();
+      fillStack.setAmount(Math.min(drained.getAmount(), 6));
+
+      // can we fill?
+      int filled = toFill.fill(fillStack, IFluidHandler.FluidAction.SIMULATE);
+      if (filled > 0) {
+        // transfer it
+        this.drained.shrink(filled);
+        fillStack.setAmount(filled);
+        toFill.fill(fillStack, IFluidHandler.FluidAction.EXECUTE);
+      }
+    }
+    else {
+      // filling TE got lost. reset. all liquid buffered is lost.
+      reset();
+    }
+  }
+
+  protected void reset() {
+    isPouring = false;
+    stopPouring = false;
+    drained = null;
+    direction = Direction.DOWN;
+    lastRedstoneState = false;
```

Comment:
Remove this last line, it led to a bug in 1.12 with block updates. We never patched it because it became expected behaior.

Reply:
aight, and its now time we can clean slate it.

Reply:
I have some plans for a proper behavior version of that basically. Changing what stops the faucet from pouring